service/api_impl.ml

```diff
@@ -94,6 +94,40 @@ let make_commit ~engine ~owner ~name hash =
           | `Failed -> Results.status_set results Failed
           | `Passed -> Results.status_set results Passed);
          Service.return response
+
+       method message_impl _params release_param_caps =
+         let open Commit.Message in
+         release_param_caps ();
+         let response, results = Service.Response.create Results.init_pointer in
+         let active =
+           Index.get_active_refs { Ocaml_ci.Repo_id.owner; name }
+           |> Index.Ref_map.bindings
+           |> List.filter_map (fun (_, (h, message, _)) ->
+                  if h = hash then Some message else None)
+         in
+         (match active with
+         | [] ->
+             ()
+             (* [benmandrew]: Theoretically this case should never happen O_O *)
```

Comment:
If it's not supposed to happen, shouldn't we use `List.hd`? @novemberkilo  what do you thing?

Reply:
Yes I think that's a good suggestion - At the moment we are swallowing the error. 

Either we should handle it by throwing an exception via `List.hd` or we use this branching approach to log an error and then manually throw an exception.

Reply:
Logging the error seems to be reasonable to me :+1: